### DISCUSSION TITLE
remark/rehype plugin output with nextMDX

### DISCUSSION THREAD

**Discussion by RobbieTheWagner** [vercel/next.js#52803] [Help]
### Summary

Is there a way to hook into the VFile and extra data from remark/rehype when using nextMDX? Specifically, I would like to use https://github.com/jrson83/rehype-extract-excerpt to make excerpts for my blog posts, but I cannot figure out where the data from remark/rehype goes or how I might access `file.data.excerpt` after the plugin runs.

**Comment by icyJoseph** [comment]
Perhaps for this is better to read the file inside GSP and pipe all the plugins you want through.

- https://nextjs.org/learn/basics/dynamic-routes/render-markdown

Like they do on that link. I ended up doing something similar once I had needs for more than what @next/mdx can do.

That being said, one can see props in mdx files by doing `{console.log(props)}` for example, or see the page props in `_app`, via `pageProps`.

**Reply by RobbieTheWagner** [reply]
@icyJoseph I don't think it's in `props` though, right? These plugins put stuff on the VFile instance under like a `data` property. There has to be a way to pass that stuff through.

**Reply by icyJoseph** [reply]
No it is not, I just mentioned for completeness sake. 

I do think that next/mdx, while useful, is never going to be as good as just compiling the MDX it yourself. It is less auto-magic as well.
